src/DGtal/shapes/Mesh.ih

```diff
@@ -285,6 +286,29 @@ DGtal::Mesh<TPoint>::getFaceColor(unsigned int i) const
 
 template<typename TPoint>
 inline
+std::pair<TPoint, TPoint>  
+DGtal::Mesh<TPoint>::getBoundingBox() const
+{
+  std::pair<TPoint, TPoint> theResult;  
+  struct CompPoints
+  {
+    CompPoints(typename TPoint::Dimension d): myDim(d){};
+    bool operator() (const TPoint &p1, const TPoint &p2){return p1[myDim]<p2[myDim];};
+    typename TPoint::Dimension myDim;
+  };
+  TPoint lowerBound, upperBound;
+  for(unsigned int i=0; i< TPoint::size(); i++)
+    {
+      upperBound[i] = (*(std::max_element(vertexBegin(), vertexEnd(), CompPoints(i))))[i];
+      lowerBound[i] = (*(std::min_element(vertexBegin(), vertexEnd(), CompPoints(i))))[i];
```

Comment:
In fact, the `static_cast` is only needed to strip away the ambiguity on the `operator[]` overloads (constant & mutable).
Nevertheless, it was not meant to be a modification proposal: I think this is completely unreadable compared to your version !

But, is it not possible to calculate the bounding box using only one pass ? Like this:
```C++
  std::pair<TPoint, TPoint> theResult( 
    TPoint::diagonal( std::numeric_limits<typename TPoint::Component>::max() ),
    TPoint::diagonal( std::numeric_limits<typename TPoint::Component>::min() )
  );

  for ( typename VertexStorage::const_iterator it = vertexBegin(), it_end = vertexEnd(); it != it_end; ++it )
    {
      theResult.first  = theResult.first.inf(*it);
      theResult.second = theResult.second.sup(*it);
    }

  return theResult;
```
supposing that `TPoint` has `inf` and `sup` method.